books/spiders/books_info.py

```diff
@@ -0,0 +1,60 @@
+from typing import Any
+import re
+
+import scrapy
+from scrapy import Selector
+from scrapy.http import Response
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+
+class BooksInfoSpider(scrapy.Spider):
+    name = "books_info"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        options = webdriver.ChromeOptions()
+        options.add_argument("--headless")
+        self.driver = webdriver.Chrome(options=options)
+
+    def close(self, reason) -> None:
+        self.driver.close()
+
+    def parse(self, response: Response, **kwargs):
+        for book in response.css(".product_pod"):
+            yield self.parse_detail_page(response, book)
+
+        next_page = response.css(".pager > .next > a::attr(href)").get()
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+    def _get_num_available_books(self, text: str) -> int:
+        match = re.search(r'\((\d+) available\)', text)
+        if match:
+            return int(match.group(1))
+        return 0
+
+    def parse_detail_page(self, response: Response, book: Selector) -> dict[str, Any]:
+        detailed_url = response.urljoin(book.css("h3 > a::attr(href)").get())
+        self.driver.get(detailed_url)
```

Comment:
Using Selenium's `get` method to navigate to a new page can be slow. If the content can be accessed directly via Scrapy, consider using Scrapy's `response.follow` instead.